client/web-sveltekit/src/lib/dom.ts

```diff
@@ -247,13 +247,15 @@ interface ComputeFitAttributes {
 export const computeFit: Action<HTMLElement, void, ComputeFitAttributes> = node => {
     // Holds the cumulative width of all elements up to element i.
     const widths: number[] = [0]
+    const parentRight = node.getBoundingClientRect().left
 
     for (let i = 0; i < node.children.length; i++) {
-        widths[i + 1] = node.children[i].getBoundingClientRect().right
+        widths[i + 1] = node.children[i].getBoundingClientRect().right - parentRight
     }
 
     function compute(): void {
-        const right = node.getBoundingClientRect().right
+        const rootRect = node.getBoundingClientRect()
+        const right = rootRect.right - rootRect.left
```

Comment:
We can probably just use `width` here.

```suggestion
        const nodeWidth = node.getBoundingClientRect().width
```